cdap-api/src/main/java/co/cask/cdap/api/schedule/PartitionTriggerInfo.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ * The dataset partition trigger information to be passed to the triggered program.
+ */
+public class PartitionTriggerInfo extends TriggerInfo {
```

Comment:
E.g. ApplicationSpecification, RunId are both interfaces, hence you need custom read/write logic for them. Also, you might want to define the `TriggerInfo` hierarchy as interfaces in order to be able to do serialization with non-api classes.